### PR TITLE
Introduce since, description and default tags

### LINKED ISSUE #251: Default, Version & Description Parameter Annotations
## 🚀 Feature Proposal

Enable the enrichment of request parameters via annotation with
* a default value
* version introduced 
* parameter description
All values should be optional and this must been seen as a long term goal to provide this data with collaboration of other teams.

## Motivation

1. Developer experience, by getting a hint on the default value of a parameter and a proper description.
2. The docs team could provide more elaborate generated documentation.
3. The UI could add more context to UI fields and generate larger parts of forms.

## Example

Example definition of a param within a request class:
```
/**
 * @default 30s
 * @description This is the description of a parameter that was introduced in 7.8.0 with the default value 30s
 * @version 7.8.0
 */
a_parameter: string
```

json output:

```
{
  "name": "a_parameter",
  "required": true,
  "default": "30s",
  "description": "This is the description of a parameter that was introduced in 7.8.0 with the default value 30s",
  "version": "7.8.0",
  "type": {
    "kind": "instance_of",
    "type": {
      "name": "SearchableSnapshotsUsage",
      "namespace": "x_pack.info.x_pack_usage"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by delvedor**
Closes #251

### CODE REVIEW

**Review by philkra — APPROVED** [review]
LGTM

File: docs/modeling-guide.md [hunk 1]
```diff
@@ -278,4 +278,67 @@ class FooContainer {
...
+
+### Additional information
+
+If needed, you can specify additional information on each type with the approariate JSDoc tag.
+Following you can find a list of the supported tags:
+
+#### `@since`
+
+Every API already has a `@since` tag, which describes when an API has been added.
+You can specify an additional `@since` tag for every parameter that has been added afterwards.
```
Comment (swallez) [inline docs/modeling-guide.md:291]: "property" would be more suited than "parameter" IMHO as this applies to path & query parameters and json properties, which are all represented as TS structure properties

File: docs/modeling-guide.md [hunk 2]
```diff
@@ -278,4 +278,67 @@ class FooContainer {
...
+```
+
+#### `@description`
+
+A custom description of the property.
+
+```ts
+class Foo {
+  bar: string
+  /** @description You can baz! */
```
Comment (swallez) [inline docs/modeling-guide.md:313]: Do we need an annotation for this ? This should be the jsdoc comment text:

```ts
/**
 * You can baz!
 */
```

And BTW we also should define the markup langage in descriptions. My suggestion is to use [CommonMark](https://commonmark.org/) that is implemented in all languages and that code generators can translate to whatever markup language is used in their respective documentation (some HTML with custom markers in JavaDocs, reStructuredText in Python, etc)

**Review by swallez — CHANGES_REQUESTED** [review]

File: docs/modeling-guide.md (hunk 2 above)
Comment (delvedor) [reply docs/modeling-guide.md:313]: Good suggestion, let me see what I can do about it.
Regarding the markup, I like the idea, but let's discuss it in a separate issue.

**Review by swallez — APPROVED** [review]
LGTM

**Review by philkra — APPROVED** [review]
LGTM - I like `server_default` much better!

**Pull request merged**